### ISSUE TITLE
ConvertFacenetTensorflowForJetsonTx1

### ISSUE THREAD

**Issue by billbak90** [davidsandberg/facenet#125]
Is it possible  to convert Facenet/TensorFlow implementation to   Caffe Model for running with TensorRT on nvidia JetsonTX1?

**Comment by davidsandberg** [comment]
I'm sure it's possible but i'm not sure it's easy. You'd probably be better off using a tool already using caffe, like [this one](https://github.com/ydwen/caffe-face).

**Comment by billbak90** [comment]
My target is to re-implement compare.py with caffe and after that to deploy it on a jetson TX1 .
Thanks for helping me !!!

**Comment by achbogga** [comment]
Hi @davidsandberg @billbak90 , I have tried this with no luck. Also, I have tried converting the current pretrained tensorflow model directly as mentioned in the docs of TensorRT 4.0.0.3. Please find the output of the conversion attached in a html file.

From what I have read and believe, there seem to be some unsupported layers in the current implementation of the facenet network. I dont have the access to the dataset used to train the model. But, @davidsandberg to you this should be easy to replace the non supported operations with the supported operations according to the [tensorRT developer guide](http://docs.nvidia.com/deeplearning/sdk/tensorrt-developer-guide/index.html) and retrain the model. Please look into this and I will highly appreciate your effort.
Thanks a million!


[facenet_replace_graph_nodes.pdf](https://github.com/davidsandberg/facenet/files/1881567/facenet_replace_graph_nodes.pdf)

**Comment by achbogga** [comment]
Hello @davidsandberg ,

I have successfully converted the entire facenet graph into tensorRT engine after removing just the batch normalization. However, I don't have access to the data to retrain the network. I would highly appreciate (I can buy you a cool gift) if you can just retrain without batch normalization and provide us the trained model file.

Thanks a million in advance.
Hoping to get a response,
Achyut Boggaram

**Comment by congphase** [comment]
> Hello @davidsandberg ,
> 
> I have successfully converted the entire facenet graph into tensorRT engine after removing just the batch normalization. However, I don't have access to the data to retrain the network. I would highly appreciate (I can buy you a cool gift) if you can just retrain without batch normalization and provide us the trained model file.
> 
> Thanks a million in advance.
> Hoping to get a response,
> Achyut Boggaram

Can you explain some details on how you have removed the batch normalization? I'm working on using this facenet to run on Jetson too, but Jetson Nano. Thank you a lot!!
